src/util.ts

```diff
@@ -67,6 +113,27 @@ export function extractPropertyEntry(
   };
 }
 
+/**
+ * Returns the class fields of a class
+ * @param {ESTree.Class} node Class to retrieve class fields for
+ * @return {ReadonlyMap<string, ESTreeObjectExpression>}
+ */
+export function getClassFields(
+  node: ESTree.Class
+): ReadonlyMap<string, ESTree.PropertyDefinition> {
+  const result = new Map<string, ESTree.PropertyDefinition>();
+
+  for (const member of node.body.body) {
+    if (
+      member.type === 'PropertyDefinition' &&
+      member.key.type === 'Identifier'
```

Comment:
this is where you would also check that `!member.declare` (since a declared class field isn't a class field in this respect i guess)